core/state/trie_prefetcher.go

```diff
@@ -77,15 +112,91 @@ func newTriePrefetcher(db Database, root common.Hash, namespace string) *triePre
 		storageWasteMeter: metrics.GetOrRegisterMeter(prefix+"/storage/waste", nil),
 	}
 	go p.abortLoop()
+	go p.mainLoop()
 	return p
 }
 
+func (p *triePrefetcher) mainLoop() {
+	for {
+		select {
+		case <-p.copyChan:
+			fetcherCopied := &triePrefetcher{
+				db:      p.db,
+				root:    p.root,
+				fetches: make(map[common.Hash]Trie, len(p.fetchers)),
+			}
+			// we're copying an active fetcher, retrieve the current states
+			for root, fetcher := range p.fetchers {
+				fetcherCopied.fetches[root] = fetcher.peek()
+			}
+			p.copyDoneChan <- fetcherCopied
```

Comment:
For channel of channel:
** pros: save channel buffer & return value in order
** cons: temporary channel variables

Accepted, since channel variable's cost is negligible,  can be seen as a normal variable.